posthog/api/batch_exports.py

```diff
@@ -0,0 +1,224 @@
+            "id",
+            "destination",
+            "schedule",
+            "runs",
+            "created_at",
+            "last_updated_at",
+        ]
+        read_only_fields = [
+            "id",
+            "schedule",
+            "runs",
+            "created_at",
+            "last_updated_at",
+        ]
+
+    def create(self, validated_data: dict) -> BatchExport:
+        """Create a BatchExport."""
+        destination_data = validated_data.pop("destination")
+        schedule_data = validated_data.pop("schedule")
+        team = Team.objects.get(id=self.context["team_id"])
+
+        destination = BatchExportDestination.objects.create(team=team, **destination_data)
+        schedule = BatchExportSchedule.objects.create(team=team, **schedule_data)
+
+        batch_export = BatchExport.objects.create(team=team, schedule=schedule, destination=destination)
+        return batch_export
+
+    def get_runs(self, batch_export):
+        serializer = BatchExportRunSerializer(BatchExportRun.objects.filter(batch_export=batch_export), many=True)
+        return serializer.data
+
+
+class BatchExportViewSet(StructuredViewSetMixin, viewsets.ModelViewSet):
+    queryset = BatchExport.objects.all()
+    permission_classes = [IsAuthenticated, ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission]
+    serializer_class = BatchExportSerializer
+
+    def get_queryset(self):
+        if not isinstance(self.request.user, User) or self.request.user.current_team is None:
+            raise NotAuthenticated()
+
+        return self.queryset.filter(team_id=self.request.user.current_team.id).prefetch_related(
+            "destination", "schedule"
+        )
+
+    @action(methods=["POST"], detail=True)
+    def backfill(self, request: request.Request, *args, **kwargs) -> response.Response:
+        """Trigger a backfill for a BatchExport."""
+        if not isinstance(request.user, User) or request.user.current_team is None:
+            raise NotAuthenticated()
+
+        start_at_input = request.data.get("start_at", None)
+        end_at_input = request.data.get("end_at", None)
+
+        start_at = dt.datetime.fromisoformat(start_at_input) if start_at_input is not None else None
+        end_at = dt.datetime.fromisoformat(end_at_input) if end_at_input is not None else None
+
+        batch_export = self.get_object()
+        batch_export.backfill(start_at, end_at)
+
+        serializer = self.get_serializer(batch_export)
+        return response.Response(serializer.data)
+
+    @action(methods=["PATCH"], detail=True)
+    def pause(self, request: request.Request, *args, **kwargs) -> response.Response:
+        """Pause a BatchExport."""
+        if not isinstance(request.user, User) or request.user.current_team is None:
+            raise NotAuthenticated()
+
+        user_id = request.user.distinct_id
+        team_id = request.user.current_team.id
+        note = f"Unpause requested by user {user_id} from team {team_id}"
```

Comment:
We have an "activity log" used for other data type changes, thinking maybe it's relevant here as well. Not important for this PR though.